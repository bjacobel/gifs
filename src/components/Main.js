import React, { Component } from 'react';

import GifColumn from './GifColumn';
import SearchBox from './SearchBox';
import LoginButton from './LoginButton';
import TagBox from './TagBox';

export default class Main extends Component {
  render() {
    return (
      <div>
        <div className="column-container">
          <div className="column gifs">
            <GifColumn />
          </div>
          <div className="column info">
            <div className="info-liner">
              <SearchBox />
              <TagBox />
            </div>
          </div>
        </div>
        <LoginButton />
      </div>
    );
  }
}
