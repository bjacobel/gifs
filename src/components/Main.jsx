import React, { Component } from 'react';
import { connect } from 'react-redux';

import { getGifsAsync } from '../actions/gifs';
import { getTagsAsync } from '../actions/tags';
import GifGrid from './GifGrid';

function mapStateToProps(state) {
  return {
    gifs: state.gifs,
    tags: state.tags
  };
}

const mapDispatchToProps = {
  getGifsAsync,
  getTagsAsync
};

class Main extends Component {
  componentDidMount() {
    this.props.getGifsAsync();
    this.props.getTagsAsync();
  }

  render() {
    const { gifs, tags } = this.props;

    return <GifGrid gifs={ gifs } tags={ tags }/>;
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Main);
