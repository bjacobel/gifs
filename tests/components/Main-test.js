import React from 'react';
import { shallow } from 'enzyme';

import Main from '../../src/components/Main';

describe('Main component', () => {
  it('matches snapshot', () => {
    expect(shallow(<Main />)).toMatchSnapshot();
  });

  it('passes search param along to SearchBox if it gets it from React Router', () => {
    const enzymeRepr = shallow(<Main params={ { search: 'foo' } } />);

    expect(enzymeRepr.find('Connect(SearchBox)').prop('initialSearch')).toEqual('foo');
  });
});
