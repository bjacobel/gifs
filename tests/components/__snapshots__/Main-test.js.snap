exports[`Main component matches snapshot 1`] = `
<div>
  <div
    className="column-container">
    <div
      className="column gifs">
      <Connect(GifColumn) />
    </div>
    <div
      className="column info">
      <div
        className="info-liner">
        <Connect(SearchBox) />
        <Connect(TagBox) />
      </div>
    </div>
  </div>
  <Connect(LoginButton) />
</div>
`;
